frontend/src/components/ModelBox/ModelConfigurationSlider.tsx

```diff
@@ -32,20 +31,18 @@ function ModelConfigurationSlider({
 	return (
 		<fieldset className="model-config-slider-fieldset">
 			<legend>{config.name}</legend>
-			<button
+			<details
 				className="info-icon prompt-injection-min-button"
```

Comment:
i have replaced it with the the same classname as the detail button in the defense-mechanism block, is that okay since are the same type of button?

Reply:
Not really, because then ModelConfigurationSlider component would be relying on styles it doesn't import.

Ideally we want a common component that both can use: then we would use the same styles (and JSX code) for both. It can be pretty simple, if the only thing in common is an enclosing details element, a summary with the text "Details", and then the children it should render. Off the top of my head, something like this:

```
import './StyledDetails.css'; // common styles for details and summary in here

function StyledDetails({ children, onToggle }: { children: ReactNode; onToggle?: () => void; }) {
  return (
    <details className="styled-details" onToggle={onToggle}>
      <summary>Details</summary>
      {children}
    </details>
  );
}
```

We would use it like this:

```
import 'ModelConfigurationSlider.css';

function ModelConfigurationSlider({ config }: { config: CustomChatModelConfiguration; }) {
  ... component stuff here ...

  return (
    <fieldset className="model-config-slider-fieldset">
      <legend>{config.name}</legend>
      <StyledDetails>
        <div className="info-text">{config.info}</div>
      </StyledDetails>
      <div className="config-slider"> ... </div>
    </fieldset>
  );
}
```

The individual config components are responsible for defining their own content styles, e.g. "info-text" class above.

If you need any help with this, please just ask.